handle a particular case where stage name is not defined

FXHibon opened the pull request:
It seems that particular stages might not have a `name` attribute. When dealing with such streams, an exception is thrown in `VizGraphProcessor`. 

This PR fix this by adding a default stage name when such situations occurs.

Example of stream that would throw exception when generating diagram.

```scala
val myFlow = Source.single("hello")
      .log("logger")
      .to(Sink.head)
```

throw this exception 

> The property does not exist as it has no key, value, or associated element
java.lang.IllegalStateException: The property does not exist as it has no key, value, or associated element
	at org.apache.tinkerpop.gremlin.structure.Property$Exceptions.propertyDoesNotExist(Property.java:151)
	at org.apache.tinkerpop.gremlin.structure.util.empty.EmptyVertexProperty.value(EmptyVertexProperty.java:74)
	at net.mikolak.travesty.VizGraphProcessor.$anonfun$toVizGraph$1(VizGraphProcessor.scala:23)

I've added a test that validate the expected behavior.

Feel free to comment or criticize this fix :)

FXHibon committed:
fix(vizGraphProcessor): handle a particular case where stage name is not defined

mikolak-net commented:
I'm very sorry, for some reason I only saw this today. Will review in the evening CEST.

mikolak-net commented on src/main/scala/net/mikolak/travesty/VizGraphProcessor.scala:
```diff
@@ -19,7 +19,8 @@ class VizGraphProcessor(typeNameSimplifier: TypeNameSimplifier) {
     val implNames = in.V().l().foldRight(Map.empty[AkkaStage, String]) { (v, map) =>
-      val baseName = v.asScala.property(properties.node.StageName).value()
+      // @fixme default stage name might not be the best solution
+      val baseName = v.asScala.property(properties.node.StageName).orElse("default stage name")
```
"default stage name" is totally fine :). However, this should be loaded from the config, optimally in a new subsection (`render` ?), since it doesn't really seem like any of the existing ones fit.

Here's the config CC:
https://github.com/mikolak-net/travesty/blob/master/src/main/scala/net/mikolak/travesty/setup/TravestyConfig.scala
and here's the wiring class where the config gets injected:
https://github.com/mikolak-net/travesty/blob/master/src/main/scala/net/mikolak/travesty/setup/Wiring.scala

You can also read about the general concept [here](https://mikołak.net/blog/2016/type-safe-config-ficus-macwire.html), but honestly, you don't really need to - it's simple enough to follow by example.

mikolak-net commented on src/test/scala/net/mikolak/travesty/VizGraphProcessorSpec.scala:
```diff
@@ -101,6 +101,17 @@ class VizGraphProcessorSpec extends FlatSpec with MustMatchers with MustVerb {
+    val output = calculateResult(input)
+
+    output.nodeList() must have size 3
+    output.linkList() must have size 2
```
Once you add the configurable default name, there should be an assertion that checks whether that name is used as well.

mikolak-net commented:
@FXHibon : are you up for doing the above? If so, great, if not, no big deal, I'll just merge as-is and add the necessary modifications myself.

FXHibon commented:
> are you up for doing the above?

absolutely yes @mikolak-net

FXHibon committed:
refactor(vizGraphProcesor): add test & minor refacto

FXHibon commented:
I've added one commit to follow your recommandations.
In the same time, I've updated with the last version of sbt and last version of scala, I hope this is not a problem for you @mikolak-net

mikolak-net commented:
Re: version bump - normally it's something for a separate PR, but let's face it, this not exactly a project with a lot of contributors, so of course it's OK :).

In general, LGTM. Thank you for your contribution. I'm merging it now, should be releasing the new version today evening.
